### PR TITLE
[GO] Command Fix

### LINKED ISSUE #1270: ExecuteCommand changes
ExecuteCommand must return EmptyResponse, Response(value), Exception
```
message CommandResponse {
  oneof kind {
    Empty no_response = 1; // Server/Device does not have a response for the command.
    Value response = 2;
    Exception exception = 3;  // This is for Server/Device error executing the command.
  }
}
```
Currently ExecuteCommand only ever returns Response and EmptyResponse (only if BL returns nil). It is missing the following to adhere to spec:
- response boolean
- exception response

## 📖 Description
response boolean: if false return empty response after handler is done and throwaway handler return
exception response: populate according to the proto definition
```
message Exception {
  string type = 1;
  PolyglotText error_message = 2;
  string details = 3;
}
```

---

## 🎯 Goal / Outcome
ExecuteCommand correctly returns EmptyResponse, Response or Exception in the correct scenarios.

---

## ✅ Definition of Done (DoD)
- [ ] Feature is implemented and works as expected
- [ ] Edge cases are handled
- [ ] Tests are added or updated (if applicable)
- [ ] Documentation is updated (if needed)
- [ ] Code is reviewed and approved

---

## 🛠️ Implementation Notes (Optional)

---

## 🔗 Related Issues / Links

---
## 🚧 Risks / Open Questions (Optional)

---

### PR DESCRIPTION
**Pull request by nelgd**
## 📖 Description

Added `CommandResult` type that wraps `*protos.CommandResponse`, following the same proto-wrapper pattern used by `CatenaDevice` and other SDK types. 

Added `PolyglotText` type mapping BCP-47 language codes to display strings.

The REST server now also supports the `respond` query parameter, allowing clients to opt out of receiving command responses.

---

## ✅ Definition of Done (DoD)
- [x] `CommandResult` type wraps `*protos.CommandResponse` with `no_response`, `response`, and `exception` variants
- [x] Helper constructors provided (`CommandReply`, `CommandNoResponse`, `CommandExceptionResult`, `CommandError`)
- [x] `GetProtoResponse()` accessor exposes the underlying `*protos.CommandResponse`
- [x] `PolyglotText` type added with `NewPolyglotText`, `With`, and `Get` methods
- [x] `ExecuteCommandHandler` signature updated to return `(CommandResult, StatusResult)` without `http.ResponseWriter`/`*http.Request` params
- [x] `respond` query parameter support added to the REST command endpoint (defaults to `true`)
- [x] `WriteCommandResponseJSON` helper added for serializing `CommandResponse` proto to JSON
- [x] Examples updated with commands demonstrating all three result types (response, no\_response, exception)
- [x] "Command not found" in examples uses transport-level `CommandError` instead of `CommandExceptionResult`
- [x] Tests added for `PolyglotText`, response value, exception response, `respond=false`/`true`, and default behavior

---

## 🛠️ Implementation Notes

- Added `command.go` — `CommandResult` wrapping `*protos.CommandResponse`, `PolyglotText` type, and helper constructors
- Modified `server.go` — removed `http.ResponseWriter`/`*http.Request` from `ExecuteCommandHandler` signature, replaced `ToProto()` with `GetProtoResponse()`, added `respond` query param logic
- Modified `json_utils.go` — added `WriteCommandResponseJSON` for marshalling `*protos.CommandResponse`
- Modified `executeCommand_REST.go` — migrated to proto-wrapper `CommandResult` API, changed "command not found" to transport-level `CommandError`, added `reset` (no\_response) and `error` (exception) example commands
- Modified `getSetValue_REST.go` — updated `ExecuteCommandHandler` signature
- Modified `server_test.go` — updated handler signatures and added tests for response/exception/respond-param behavior
- Modified `command_test.go` — updated tests for proto-wrapper pattern and added `PolyglotText` tests

---

## 🔗 Related Issues / Links

Closes issue #1270

### CODE REVIEW

**Comment by nelgd** [comment]
bugbot run

**Review by ctwarog-ross — CHANGES_REQUESTED** [review]
While this solution works, it doesn't quite follow the pattern of other types like for example device, which is just a wrapper of the proto equivalent. Wrapping the proto hides it from business logic but allows the sdk to use all the logic of the compiled protos, as well as theoretically allowing for more seamless accomodation for changes to the protos should that occur.

File: sdks/go/examples/executeCommand_REST/executeCommand_REST.go [hunk 1]
```diff
@@ -232,20 +232,17 @@ func main() {
...
 			logger.Warning("Command not found", "slot", slot, "command", commandFqoid)
-			exception := map[string]any{
-				"exception": map[string]any{
-					"type":    "Invalid Command",
-					"details": "Command not found: " + commandFqoid,
-				},
-			}
-			val, _ := catena.ToCatenaValue(exception)
-			return catena.Reply(val)
+			return catena.CommandExceptionResult(
```
Comment (abrown-RV) [inline sdks/go/examples/executeCommand_REST/executeCommand_REST.go]: This should be a transport level error, not a CommandException

File: sdks/go/examples/executeCommand_REST/executeCommand_REST.go [hunk 2]
```diff
@@ -174,45 +174,45 @@ func main() {
...
 		// Reset command
-		"reset": func(payload any) (catena.CatenaValue, catena.StatusResult) {
+		"reset": func(payload any) (catena.CommandResult, catena.StatusResult) {
 			counter.Reset()
 			logger.Info("Counter reset", "value", counter.GetValue())
 			val, _ := catena.ToCatenaValue(buildResponse())
-			return catena.Reply(val)
+			return catena.CommandReply(val)
 		},
 	}
```
Comment (abrown-RV) [inline sdks/go/examples/executeCommand_REST/executeCommand_REST.go:227]: Can we get example commands that return a non response, and an exception?

File: sdks/go/pkg/catena/command.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+// CommandNoResponse returns an empty command response (no_response).
+func CommandNoResponse() (CommandResult, StatusResult) {
+	return CommandResult{kind: commandResultEmpty}, StatusResult{Code: OK}
+}
+
+// CommandExceptionResult returns a command exception response.
+// exType is the exception type, details provides additional context,
+// and errorMessage is a map of language code to display string (may be nil).
+func CommandExceptionResult(exType, details string, errorMessage map[string]string) (CommandResult, StatusResult) {
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/command.go]: We're going to want a PolyglotText type eventually, may as well do it now. Some code suggestions from copilot:
```go
// PolyglotText maps BCP-47 language codes to display strings.
type PolyglotText map[string]string

// NewPolyglotText creates a PolyglotText with a single language entry.
func NewPolyglotText(lang, text string) PolyglotText {
    return PolyglotText{lang: text}
}

// With returns the PolyglotText with an additional language entry added.
func (p PolyglotText) With(lang, text string) PolyglotText {
    p[lang] = text
    return p
}

// Get returns the display string for lang, falling back to fallback if not present.
func (p PolyglotText) Get(lang, fallback string) string {
    if s, ok := p[lang]; ok {
        return s
    }
    return p[fallback]
}


// single language
NewPolyglotText("en", "Command not found")

// multiple languages
NewPolyglotText("en", "Command not found").With("fr", "Commande introuvable")
```

File: sdks/go/pkg/rest/json_utils.go [hunk 4]
```diff
@@ -50,6 +50,33 @@ import (
 	"github.com/rossvideo/catena/build/go/protos"
 )
 
+// WriteCommandResponseJSON marshals a protos.CommandResponse to JSON and writes it
+// to the HTTP response with the specified status code.
+func WriteCommandResponseJSON(w http.ResponseWriter, cmdResp *protos.CommandResponse, statusCode int) error {
+	w.Header().Set("Content-Type", "application/json")
+
+	if cmdResp == nil {
+		w.WriteHeader(statusCode)
```
Comment (abrown-RV) [inline sdks/go/pkg/rest/json_utils.go:59]: hit these err branches if you can

File: sdks/go/pkg/rest/server.go [hunk 5]
```diff
@@ -62,7 +62,7 @@ type DeviceHandler func() (catena.CatenaDevice, catena.StatusResult)
 type GetValueHandler func(slot int, fqoid string) (catena.CatenaValue, catena.StatusResult)
 type SetValueHandler func(value any, slot int, fqoid string) catena.StatusResult
 type GetAssetHandler func(slot int, fqoid string) (catena.CatenaAsset, catena.StatusResult)
-type ExecuteCommandHandler func(w http.ResponseWriter, r *http.Request, slot int, commandFqoid string, payload any) (catena.CatenaValue, catena.StatusResult)
+type ExecuteCommandHandler func(w http.ResponseWriter, r *http.Request, slot int, commandFqoid string, payload any) (catena.CommandResult, catena.StatusResult)
```
Comment (abrown-RV) [inline sdks/go/pkg/rest/server.go]: can take out the http references from the handler now

**Review by abrown-RV — DISMISSED** [review]

**Review by ctwarog-ross — COMMENTED** [review]
This is exactly what I wanted to see. Just needs rebased and fixed with the changes made in the grpc merge (apologies).

**Review by ctwarog-ross — APPROVED** [review]
